### ISSUE TITLE
`currently` block often struggles with precipitation

### ISSUE THREAD

**Issue by cloneofghosts** [Pirate-Weather/pirateweather#390]
### Describe the bug

I've noticed that while the API is usually pretty good about showing rain it often struggles with snow. Today it's been snowing since about 1:30pm the API has only shown the snow icon once. It was at one point briefly going to show rain but then the models updated and the API was no longer forecasting rain.

I'm not sure if this is something with HRRR or something on PWs end. I did check SpotWX and HRRR is forecasting snow for the next few hours and I checked the subhourly HRRR graphics and the radar does show precipitation in the area.

### Expected behavior

API should show `snow` precipitation when its snowing

### Actual behavior

API often struggles with `snow` precipitation when its snowing

### API Endpoint

Production

### Location

Ottawa, Ontario

### Other details

I also think that the API struggles with `sleet` as well though it comes up less often than snow so is harder to notice.

### Troubleshooting steps

- [X] I have searched this repository and [Home Assistant Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have read through the [API documentation](https://pirateweather.net/en/latest/API/) before opening this issue.
- [X] I have not included my API key anywhere in my issue.
- [X] I have written an informative title.

**Comment by cloneofghosts** [comment]
@alexander0042 Any updates on this? There's been a few occasions where it's been snowing (albeit very light snow) and the API hasn't been showing any snow. There's some very light snow occurring at the moment and the API isn't forecasting any snow (it could be because it just started recently?).

When I look at the forecast the only model forecasting snow for today is the GFS/GEFS model and others are forecasting nothing. I checked a few other sites and Accuweather and ECCC are forecasting snow whereas TWC (weather.com) is not.

**Comment by cloneofghosts** [comment]
One day after I complained about this the API is now showing very light snow (`"precipIntensity": 0.036`) in my location which is accurate. It seems to me like HRRR can't show precipitation between 0.036 mm/h and 0.36 mm/h? I know at one point this was 0.25 mm/h but it changed at some point?

It could be that this is just an HRRR issue which can't be fixed on PWs end which in that case hopefully RRFS can fix things when its launched next year (I saw somewhere that it'll be back online in a few months but wont fully launch until 2026)

**Comment by cloneofghosts** [comment]
I had initially posted this about the API missing `snow` precipitation it seems like (today at least) it's also missing rain. The API showed rain about 1-2 hours after it started and it went back to showing no precipitation about 10-20m ago while its still lightly raining.

I also noticed at night the temperatures seem to be way off the temperatures the airport was showing. I think it was around 10pm yesterday where PW was showing -1C and the airport was around 5C. Also seems a bit off today as well though not as bad.

@alexander0042 Is this just the HRRR `precipRate` variable missing precipitation or is it something else?

**Comment by alexander0042** [comment]
A couple thoughts here:
1. Visibility in HRRR subh must be something new, so that's a good addition! Visibility ends up being used for a bunch of stuff, so would love to have higher res data here.
2. Clarification question here, and I love the Panoply plots to debug it (that's exactly what I use too!). This seems to be an issue mostly with currently data, correct? Is the precipitation showing up in the hourly data? Isolate what model this is stemming from

**Comment by cloneofghosts** [comment]
While not related to this issue but is related to precipitation so will add here. I can always create a separate issue if you'd like.

When you're zeroing out the precipitation with no probability (or probability < 5% since I think you zero it if below that you should also apply it to minutely/currently. The accumulation should also be zeroed or if you'd prefer to keep things closer to the raw data add a check if pop = 0 (or <= 5%) so it doesn't affect the summaries.

**Comment by cloneofghosts** [comment]
@alexander0042 Having this issue again this afternoon. It's been raining/snowing since about noon and the API stopped showing any precipitation about 1-2 hours ago. I can look at Panoply to show the reflectivity product if you want as I find that sometimes reflectivity will show precipitation in the area whereas precip rate doesn't show anything.

**Comment by alexander0042** [comment]
Here's an interesting test- can you try checking the "currently" value for these times now, using the regular API endpoint? It'd be interesting to see if it's showing precipitation now, as opposed to in the moment

**Comment by cloneofghosts** [comment]
> Here's an interesting test- can you try checking the "currently" value for these times now, using the regular API endpoint? It'd be interesting to see if it's showing precipitation now, as opposed to in the moment

Checked the API for 3PM, 4PM, 5PM and 6PM yesterday and the API is showing snow in the currently block but didn't yesterday afternoon.

**Comment by cloneofghosts** [comment]
I know that the subhourly HRRR has reflectivity products available which might be useful here? You can convert dBZ to mm/h using this formula: 0.036 x 10^(0.0625 x dBZ) (formula found [here](https://community.windy.com/post/130217)). Anything below 5 dBZ in't classified as any precipitation.

**Comment by alexander0042** [comment]
That's a great idea too! I need to put radar reflectivity in anyway to get the maps added in so this is something that needs to be done regardles

**Comment by cloneofghosts** [comment]
@alexander0042 Any updates here? I know winter is pretty much over but this would be useful for other types of precipitation and not just snow.

**Comment by cloneofghosts** [comment]
Saw a similar issue yesterday when it was raining. @alexander0042 any updates on this? Haven't heard anything on this since your last reply and the dev endpoint is the same as live.

**Comment by alexander0042** [comment]
Ok, I still like the idea of using accumulation here as a stepping stone while sorting out reflectivity longer term, the issue is that it's not currently being read in by subhourly. That makes it a bit of a larger change, but at least I can do all three variables (vis/ reflectivity/ accumulation) in one, which is great. 

This is mostly an implementation reminder for me, since we've never actually had to change the format of the underlying files. I knew it would happen at some point, but isn't particularly elegant: 
1. Create a new SubH ingest task that reads the new updated script
2. Have it run in parallel with the existing SubH ingest
3. Have it save files with a new name
4. Edit the prod endpoint to start saving this file to the drive, along with the other one. Keep prod using old version
5. Set dev endpoint to read the new file!
6. When ready, change prod to use new file, kill old ingest task, remove old S3 copy

**Comment by cloneofghosts** [comment]
I'm converting this issue into a general precipitation issue since it seems to not just affect snow but other precipitation types as well. @alexander0042 Any updates on this?

It's been mostly fine today but it did predict some heavy rain which didn't happen and has a few times not shown the current condition as rain.

**Comment by cloneofghosts** [comment]
@alexander0042 Seeing this again today. There's been light rain for most of the day but when I look at the API it has mostly shown cloudy conditions which I assume is due to the precipitation rate being below 0.36 mm/h?

API conditions in HA (only updates every 30m):
![Image](https://github.com/user-attachments/assets/9c37675e-bd45-49d5-a4ab-d15e61390277)

Airport conditions:
![Image](https://github.com/user-attachments/assets/30563f5d-3181-4112-a60e-f5392257aef9)

**Comment by cloneofghosts** [comment]
@alexander0042 Any ETA on when this might be fixed? The other day when it was raining the API showed cloudy conditions.

**Comment by alexander0042** [comment]
Yes! Sorry I've been quiet lately, but investigating this (and the current Time Machine hiccups) tomorrow

**Comment by cloneofghosts** [comment]
> Yes! Sorry I've been quiet lately, but investigating this (and the current Time Machine hiccups) tomorrow

No worries. Whenever you go quiet I just assume that you've been busy with life or work and if anything is urgent you'll try to look into it when available.

**Comment by cloneofghosts** [comment]
@alexander0042 Any updates for this? It's been raining for at least the last hour but the API is not showing any rain.

**Comment by cloneofghosts** [comment]
Got the currently/minutely block setup to use reflectivity on dev and precipitation is much better now. While not 100% perfect it's better than what it was before.
